src/jace/util/traits.py

```diff
@@ -0,0 +1,137 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Contains all traits function needed inside JaCe."""
+
+from __future__ import annotations
+
+from typing import Any, TypeGuard
+
+import dace
+import jax
+import numpy as np
+from jax import _src as jax_src, core as jax_core
+from jaxlib import xla_extension as jax_xe
+
+import jace.util as util
+from jace import stages
+
+
+def is_jaceified(obj: Any) -> TypeGuard[stages.JaCeWrapped]:
+    """Tests if `obj` is decorated by JaCe.
+
+    Similar to `is_jaxified` but for JaCe objects.
+    """
+
+    if util.is_jaxified(obj):
+        return False
+    return isinstance(obj, stages.JaCeWrapped)
+
+
+def is_jaxified(
+    obj: Any,
+) -> TypeGuard[jax_core.Primitive | jax_src.pjit.JitWrapped | jax_xe.PjitFunction]:
```

Comment:
I think it would be better to define a type alias (e.g. `JaxObject` or `JaxifiedObject` for this union